realm-transformer/src/main/kotlin/io/realm/transformer/build/BuildTemplate.kt

```diff
@@ -107,51 +111,46 @@ abstract class BuildTemplate(val metadata: ProjectMetaData, val outputProvider:
             BytecodeModifier.addRealmAccessors(it)
             BytecodeModifier.addRealmProxyInterface(it, classPool)
             BytecodeModifier.callInjectObjectContextFromConstructors(it)
+
+            processedClasses[it.name] = it
         }
     }
 
     abstract fun transformDirectAccessToModelFields()
 
-    fun copyResourceFiles() {
-        copyResourceFiles(inputs)
-        classPool.close();
+    fun copyProcessedClasses() {
+        for ((fqname: String, clazz: CtClass) in processedClasses) {
+            outputProvider.putNextEntry(JarEntry("${fqname.replace('.', '/')}.class"))
```

Comment:
Does this also work on Windows or should we use File.separatorChar?

Reply:
I would argue heavily that the Zip file entry header should be platform agnostic ... otherwise you would need to supply different Zip-files for each platform. And since it works on one platform then I assume that it would also work on the others.